drivers/builtin/src/pk.c

```diff
@@ -1229,13 +1252,54 @@ int mbedtls_pk_sign_restartable(mbedtls_pk_context *ctx,
     (void) rs_ctx;
 #endif /* MBEDTLS_ECP_RESTARTABLE */
 
-    if (ctx->pk_info->sign_func == NULL) {
-        return MBEDTLS_ERR_PK_TYPE_MISMATCH;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_status_t status = psa_get_key_attributes(ctx->priv_id, &attributes);
+    if (status != PSA_SUCCESS) {
+        return PSA_PK_TO_MBEDTLS_ERR(status);
     }
 
-    return ctx->pk_info->sign_func(ctx, md_alg,
-                                   hash, hash_len,
-                                   sig, sig_size, sig_len);
+#if defined(PSA_HAVE_ALG_ECDSA_SIGN) || defined(PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_BASIC)
+    psa_key_type_t type = psa_get_key_type(&attributes);
+    psa_algorithm_t hash_alg = mbedtls_md_psa_alg_from_type(md_alg);
+#endif /* defined(PSA_HAVE_ALG_ECDSA_SIGN) || defined(PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_BASIC) */
+
+#if defined(PSA_HAVE_ALG_ECDSA_SIGN)
+    if (PSA_KEY_TYPE_IS_ECC(type)) {
+        size_t key_bits = psa_get_key_bits(&attributes);
+        psa_reset_key_attributes(&attributes);
```

Comment:
I know that calling `psa_reset_key_attributes()` nowadays is no more mandatory, but since you are calling it if `PSA_KEY_TYPE_IS_ECC(type)` or `PSA_KEY_TYPE_IS_RSA(type)`, I noticed it wouldn't be called if you don't fall in one of the above mentioned conditions.
IMO for sake of uniformity, it's better to call `psa_reset_key_attributes()` at the end of the function anyway. 
My goal is either to call it in all possible code branches, or never call it.